ui/app/adapters/kv/data.js

```diff
@@ -54,24 +54,45 @@ export default class KvDataAdapter extends ApplicationAdapter {
         };
       })
       .catch((errorOrResponse) => {
+        const errorCode = errorOrResponse.httpStatus;
         // if it's a legitimate error - throw it!
-        if (errorOrResponse instanceof AdapterError) {
+        if (errorOrResponse instanceof ControlGroupError) {
           throw errorOrResponse;
         }
-        // in the case of a deleted/destroyed secret the API returns a 404 because { data: null }
-        // however, there could be a metadata block with important information like deletion_time
-        // handleResponse below checks 404 status codes for metadata and updates the code to 200 if it exists.
-        // we still end up in the good ol' catch() block, but instead of a 404 adapter error we've "caught"
-        // the metadata that sneakily tried to hide from us
-        return {
-          ...errorOrResponse,
-          data: {
-            ...errorOrResponse.data, // includes the { metadata } key we want
-            id,
-            backend,
-            path,
-          },
-        };
+
+        if (errorCode === 403) {
```

Comment:
Previously on KV, the adapter would throw the error and then manually push a record with `failedServerRead` from within the route. I moved that logic into here so we don't have to do manual pushes, but I'm curious what y'all think about this approach (since it's rather duplicative and a lot of logic). Wrote lots of tests around it though!

Reply:
I wonder if it'd clean up the logic to make `{ id, backend, path}` a variable that is spread to each response for consistency?

Mostly because I noticed `version` is included in both of the error responses below - is that intentional? It's the version queryParam which seems like something we don't want to manually add to the response

Reply:
Yeah, I like having the base items. 
I added version because if it's provided, the API is requesting that version and so the version should match (especially since it's included in the ID). If the version is not provided it won't populate on the model